src/Components/Client.js

```diff
@@ -1,85 +1,88 @@
 import io from "socket.io-client";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import Chat from "./Chat";
 import TextField from "@mui/material/TextField";
 import Button from "@mui/material/Button";
-import FileCopyIcon from "@mui/icons-material/FileCopyOutlined";
-import SaveIcon from "@mui/icons-material/Save";
-import PrintIcon from "@mui/icons-material/Print";
-import ShareIcon from "@mui/icons-material/Share";
+import Snackbar from "@mui/material/Snackbar";
+import IconButton from "@mui/material/IconButton";
 import SpeedDialAction from "@mui/material/SpeedDialAction";
 import AddBoxIcon from "@mui/icons-material/AddBox";
 import IndeterminateCheckBoxIcon from "@mui/icons-material/IndeterminateCheckBox";
+import MenuItem from "@mui/material/MenuItem";
 import {
 	Box,
 	Container,
 	createTheme,
+	FormControl,
+	InputLabel,
+	Select,
 	SpeedDial,
 	SpeedDialIcon,
+	Stack,
 	ThemeProvider,
 	Typography,
 } from "@mui/material";
-const socket = io.connect("http://localhost:3001");
+import CloseIcon from "@mui/icons-material/Close";
+var socket;
+
+/**
+ * Allows for Creation of Chats and Chat box
+ * @returns html for chat box in bottom left
+ */
 function Client() {
+	useEffect(() => {
+		socket = io.connect("http://localhost:3001");
+		return;
+	}, []);
 	const actions = [
 		{ icon: <AddBoxIcon />, name: "Join/Create Chat Room" },
 		{ icon: <IndeterminateCheckBoxIcon />, name: "Leave Chat Room" },
 	];
-	var chatrooms = [];
-	var chatroomObject = { socket: null, username: null, room: null };
-
+	var [chatrooms, updateChatrooms] = useState([]);
```

Comment:
I was under the impression that all useState ~~elements~~ ~~variables~~ _things_ are supposed to be declared as constant. 
```suggestion
	const [chatrooms, updateChatrooms] = useState([]);
```